src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -0,0 +1,395 @@
+	 * and citus_shards and finally joined by the table obtained by the grouping of colocation id and shardminvalue and shardsize exceeding the threshold.*/
+
+	appendStringInfo(
+		query,
+		" SELECT cs.shardid,pd.shardminvalue,pd.shardmaxvalue,cs.shard_size,pn.nodeid,ct.distribution_column,ct.table_name,cs.shard_name,(SELECT relname FROM pg_class WHERE oid = ct.table_name)"
+		" FROM pg_catalog.pg_dist_shard pd JOIN pg_catalog.citus_shards cs ON pd.shardid = cs.shardid JOIN pg_catalog.pg_dist_node pn ON cs.nodename = pn.nodename AND cs.nodeport= pn.nodeport"
+		" JOIN"
+		" ( select shardid , max_size from (SELECT distinct first_value(shardid) OVER w as shardid, sum(shard_size) OVER (PARTITION BY colocation_id, shardminvalue) as total_sum, max(shard_size) OVER w as max_size"
+		" FROM citus_shards cs JOIN pg_dist_shard ps USING(shardid)"
+		" WINDOW w AS (PARTITION BY colocation_id, shardminvalue ORDER BY shard_size DESC) )as t where total_sum >= %lu )"
+		" AS max_sizes ON cs.shardid=max_sizes.shardid AND cs.shard_size = max_sizes.max_size JOIN citus_tables ct ON cs.table_name = ct.table_name AND pd.shardminvalue <> pd.shardmaxvalue AND pd.shardminvalue <> ''",
+		MaxShardSize * 1024
+		);
+
+	ereport(LOG, errmsg("%s", query->data));
+	Oid shardTransferModeOid = PG_GETARG_OID(0);
+	Datum enumLabelDatum = DirectFunctionCall1(enum_out, shardTransferModeOid);
+	char *shardSplitMode = DatumGetCString(enumLabelDatum);
+	ereport(LOG, errmsg("%s", shardSplitMode));
+	if (SPI_connect() != SPI_OK_CONNECT)
+	{
+		elog(ERROR, "SPI_connect to the query failed");
+	}
+	if (SPI_exec(query->data, 0) != SPI_OK_SELECT)
+	{
+		elog(ERROR, "SPI_exec for the execution failed");
+	}
+
+	SPITupleTable *tupletable = SPI_tuptable;
+	int rowCount = SPI_processed;
+	bool isnull;
+
+	for (int rowIndex = 0; rowIndex < rowCount; rowIndex++)
+	{
+		ShardInfoData shardinfo;
+		HeapTuple tuple = tupletable->vals[rowIndex];
+
+		Datum shardIdDatum = SPI_getbinval(tuple, tupletable->tupdesc, 1, &isnull);
+		shardinfo.shardId = DatumGetInt64(shardIdDatum);
+
+		Datum shardSizeDatum = SPI_getbinval(tuple, tupletable->tupdesc, 4, &isnull);
+		shardinfo.shardSize = DatumGetInt64(shardSizeDatum);
+
+		Datum nodeIdDatum = SPI_getbinval(tuple, tupletable->tupdesc, 5, &isnull);
+		shardinfo.nodeId = DatumGetInt32(nodeIdDatum);
+
+		char *shardMinVal = SPI_getvalue(tuple, tupletable->tupdesc, 2);
+		shardinfo.shardMinValue = strtoi64(shardMinVal, NULL, 10);
+
+		char *shardMaxVal = SPI_getvalue(tuple, tupletable->tupdesc, 3);
+		shardinfo.shardMaxValue = strtoi64(shardMaxVal, NULL, 10);
+
+		shardinfo.distributionColumn = SPI_getvalue(tuple, tupletable->tupdesc, 6);
+		shardinfo.tableName = SPI_getvalue(tuple, tupletable->tupdesc, 7);
+
+		shardinfo.shardName = SPI_getvalue(tuple, tupletable->tupdesc, 9);
```

Comment:
I'm a bit confused, isn't column 9 the relname from pg_class (unused now), and column 8 the shard name?

Reply:
yes you are right column 8 shardname was appended with schemaname and for column 9 i later append it with shardid to get the shardname which i can pass  to the query as a parameter, there was no use of column 8 so didnt store it